pkg/testsuite/mock/network.go

```diff
@@ -153,8 +156,10 @@ func (e *Endpoint) Send(packet proto.Message, to ...peer.ID) {
 			e.network.dispatchersMutex.RLock()
 			defer e.network.dispatchersMutex.RUnlock()
 
-			if err := dispatcher.handler(e.id, packet); err != nil {
-				fmt.Println(e.id, "ERROR: ", err)
+			if dispatcher.handler != nil {
```

Comment:
It is not an error. We set the handler to nil upon calling "UnregisterProtocol" which is currently called when shutting down.

In theory, nothing should arrive at the network level post shutdown as we should kill all the background processes and wait for all the workers to finish their jobs but since some background workers (i.e. the scheduler) are using goroutines, we can not reliably wait for these things and in practice it can happen that the network sees an attempt to send something post shutdown (this mostly happened in artificial tests that just called defer Shutdown on the testframework, which means that we dont even have a real synchronized shutdown logic).

In general, since the mocked network is somewhat independent of the implementations that build on top of this, I think it makes sense to gracefully handle the case of an unregistered protocol and simply handle it as a noop and prevent nil-pointer panics from happening.

I am not sure if it makes sense to add any logging here as we are talking about a case that happens post shutdown and I am not sure if it makes sense to print things after the test already finished (defer).